src/app/services/post.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Post} from "../post.model";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class PostService {
+    private postsUrl = 'http://localhost:8080/post';
+
+    constructor(private http: HttpClient) {
+    }
+
+    getPosts(): Observable<any[]> {
+        const url = `${this.postsUrl}/all`;
+        return this.http.get<any[]>(url);
```

Comment:
```suggestion
        return lastValueFrom(this.http.get<any[]>(url));
```
This Observable returns only once, so it can safely be converted to a Promise using `lastValueFrom` (if I recall correctly the function name). Promises can be awaited and lead to cleaner "linear" code.

Additionally, please provide a proper type instead of `any`. This will provide type-checking and autocompletion all the way down to the template. This might be an oversight since the `addPost` function below is properly typed.

You can also omit the function return type. It is clear from the `this.http.get<>` call what gets returned (and TypeScript is perfectly fine with it too).